x/upgrade/abci.go

```diff
@@ -22,7 +24,20 @@ import (
 // skipUpgradeHeightArray is a set of block heights for which the upgrade must be skipped
 func BeginBlocker(k keeper.Keeper, ctx sdk.Context, _ abci.RequestBeginBlock) {
 	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
 	plan, found := k.GetUpgradePlan(ctx)
+
+	if !DowngradeVerified {
+		DowngradeVerified = true
+		lastAppliedPlan, _ := k.GetLastCompletedUpgrade(ctx)
+		// verify if we are running valid app version
+		if !found || !plan.ShouldExecute(ctx) || (plan.ShouldExecute(ctx) && k.IsSkipHeight(ctx.BlockHeight())) {
```

Comment:
Doesn't block merge: Why is !found here? Doesn't that mean that if an upgrade is scheduled, we just won't do the downgrade check?